modules/core/src/lib/layer.ts

```diff
@@ -746,7 +746,24 @@ export default abstract class Layer<PropsT extends {} = {}> extends Component<
   /** (Internal) Sets the picking color at the specified index to null picking color. Used for multi-depth picking.
      This method may be overriden by layer implementations */
   disablePickingIndex(objectIndex: number): void {
-    this._disablePickingIndex(objectIndex);
+    // When rendered by a composite layer, the objectIndex may be indexed by all data in the composite layer, but this
+    // layer only renders a subset. So we allow the parent layer to translate the index into the correct one for this layer.
+    if (this.parent) {
+      const transformedIndex = this.parent._getSublayerPickingIndex(objectIndex);
```

Comment:
I am not a fan of this solution. It adds yet another hidden API and complicates the interaction between a composite layer and its sub layers.

The issue you are trying to address here is that `objectIndex` does not map to the index into `props.data`. The problem exists because picking colors in the binary GeoJSON case is recalculated and passed down to the sub layers via `data.attribute.instancePickingColors`. You can resolve the index transform locally by searching within this value:

```js
disablePickingIndex(objectIndex: number) {
  const {pickingColors, instancePickingColors} = this.getAttributeManager().attributes;
  const colors = pickingColors || instancePickingColors;
  const {attributes, length} = this.props.data;
  const externalColorAttribute = colors && attributes && attributes[colors.id];
  if (externalColorAttribute) {
    const values = externalColorAttribute.value;
    const objectColor = this.encodePickingColor(objectIndex);
    for (let index = 0; index < length; index ++) {
      const i = colors.getVertexOffset(index);
      if (values[i] === objectColor[0] &&
        values[i + 1] === objectColor[1] &&
        values[i + 2] === objectColor[2]) {
          objectIndex = index;
          break;
      }
    }
  }
  this._disablePickingIndex(objectIndex);
}

Reply:
would you prefer to put that logic here (in layer.ts) or distribute it between the path and solid polygon layers (akin to the current handling in those layers for when they are used in the geojson layer: https://github.com/visgl/deck.gl/blob/master/modules/layers/src/path-layer/path-layer.ts#L292-L300)

Reply:
It can go into layer.ts.

Reply:
Updated to search for matching values (can't exit early from the search loop -- that doesn't work for multi polygons/lines) when there is a picking color attribute defined